### PR TITLE
feat: Support setting output file type in image compression process

### PR DESCRIPTION
**Pull request by HoonBaek**
[CLNP-2718](https://sendbird.atlassian.net/browse/CLNP-2718)
[SBISSUE-14611](https://sendbird.atlassian.net/browse/SBISSUE-14611)

### Feature & ChangeLog
* Add `outputFormat` to the image compression options
  ```tsx
  <SendbirdProvider
    ...
    imageCompression={{
      outputFormat: 'preserve' | 'png' | 'jpeg',
    }}
  >
  </SendbirdProvider>
  ```

[CLNP-2718]: https://sendbird.atlassian.net/browse/CLNP-2718?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[SBISSUE-14611]: https://sendbird.atlassian.net/browse/SBISSUE-14611?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/utils/compressImages.ts [hunk 1]
```diff
@@ -51,16 +53,21 @@ export const compressImage = ({
 
       ctx.drawImage(image, 0, 0, targetResizingWidth, targetResizingHeight);
 
+      // Change the file.name & file.type for converting file type
+      const targetFileType = outputFormat === 'preserve' ? imageFile.type : `image/${outputFormat}`;
+      const originalSubtype = imageFile.type.split('/').pop();
+      const newSubtype = targetFileType.split('/').pop();
+      const newName = imageFile.name.replace(originalSubtype, newSubtype);
```
Comment (bang9) [inline src/utils/compressImages.ts:60]: Is this logic intended to change the file extension included in the file name? It seems somewhat unsafe.

Comment (bang9) [reply src/utils/compressImages.ts:60]:
```js
'png_file.png'.replace('png', 'jpeg')
// 'jpeg_file.png'
```

Comment (HoonBaek) [reply src/utils/compressImages.ts:60]: aha... yes, good catch! let me fix it

File: src/utils/compressImages.ts [hunk 2]
```diff
@@ -51,16 +53,21 @@ export const compressImage = ({
 
       ctx.drawImage(image, 0, 0, targetResizingWidth, targetResizingHeight);
 
+      // Change the file.name & file.type for converting file type
+      const targetFileType = outputFormat === 'preserve' ? imageFile.type : `image/${outputFormat}`;
+      const originalSubtype = imageFile.type.split('/').pop();
```
Comment (bang9) [inline src/utils/compressImages.ts:58]: how about `originalFormat`?

File: src/utils/compressImages.ts (hunk 1 above)
Comment (bang9) [inline src/utils/compressImages.ts:60]: How about using `targetFile` instead of `new` to maintain context?

Comment (bang9) [reply src/utils/compressImages.ts:60]: from Copilot
```ts
function replaceExtension(filename, newExtension) {
  const dotIndex = filename.lastIndexOf('.');
  if (dotIndex === -1) {
    // No extension found, return the original filename
    return filename;
  }
  // Replace the old extension with the new one
  return filename.substring(0, dotIndex) + '.' + newExtension;
}
```

**Pull request merged**